staging/src/k8s.io/apiserver/pkg/cel/environment/environment.go

```diff
@@ -267,7 +267,10 @@ func (e *EnvSet) filterAndBuildOpts(base *cel.Env, compatVer *version.Version, h
 
 	if len(declTypes) > 0 {
 		provider := apiservercel.NewDeclTypeProvider(declTypes...)
-		providerOpts, err := provider.EnvOptions(base.TypeProvider())
+		if compatVer.AtLeast(version.MajorMinor(1, 31)) {
+			provider.SetRecognizeKeywordAsFieldName(true)
+		}
+		providerOpts, err := provider.EnvOptions(base.CELTypeProvider())
```

Comment:
It looks like this enables the feature for both new and storage expressions in 1.31. But that would mean if a 1.31 cluster uses the new feature and is rolled back to 1.30, it would break?

Reply:
In Kubernetes release version, this PR will not enable support on new expression in Kubernertes 1.31 release(only on storage expression). And the enablement of the feature will be on for kubenernetes 1.32 release after compatible version for cel env updated